harvester_e2e_tests/integrations/test_4_vm_backup_restore.py

```diff
@@ -626,6 +626,210 @@ def test_restore_replace_with_vm_shutdown_command(
         )
 
 
+@pytest.mark.p0
+@pytest.mark.backup_target
+@pytest.mark.parametrize(
+    "backup_config", [
+        pytest.param("S3", marks=pytest.mark.S3),
+        pytest.param("NFS", marks=pytest.mark.NFS)
+    ],
+    indirect=True)
+class TestBackupRestoreWithSnapshot:
+
+    @pytest.mark.dependency()
+    def test_connection(self, api_client, backup_config, config_backup_target):
+        code, data = api_client.settings.backup_target_test_connection()
+        assert 200 == code, f'Failed to test backup target connection: {data}'
+
+    @pytest.mark.dependency(depends=["TestBackupRestoreWithSnapshot::test_connection"], param=True)
+    def tests_backup_vm(self, api_client, wait_timeout, backup_config, base_vm_with_data):
+        unique_vm_name = base_vm_with_data['name']
+
+        # Create backup with the name as VM's name
+        code, data = api_client.vms.backup(unique_vm_name, unique_vm_name)
```

Comment:
We use `api_client.vms.backup(unique_vm_name, unique_vm_name)` many times in this suite.
It's ok if previous one teardown successfully but will broken if not.
May consider a unique backup name.

![image](https://github.com/user-attachments/assets/43e63c05-f28b-4eff-b738-b725cd98bf65)